### PR TITLE
Refactor history.replaceState usage

### PR DESCRIPTION
**Pull request by Soviut**
### Changes

Removed the `window.location.replaceState()` from the plugin initialization when checking the session.
The browser URL is no longer replaced with `/` when refreshing the browser.

### References

https://github.com/auth0/auth0-vue/issues/88

### Testing

- [ ] This change adds unit test coverage
- [ ] This change adds integration test coverage
- [ ] This change has been tested on the latest version of the platform/language

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [x] All code quality tools/guidelines have been run/followed

### CODE REVIEW

**Comment by frederikprijck** [comment]
As mentioned in https://github.com/auth0/auth0-vue/issues/88, we might need to fix this differently, but most likely not just get rid of it.

**Comment by Soviut** [comment]
@frederikprijck what is the actual purpose of the redirect in this case?

**Comment by Soviut** [comment]
@frederikprijck I've added your suggestion.

**Comment by Soviut** [comment]
One thing I've realized, isn't the router redirect that happens inside the client proxy `handleRedirectCallback()` going to conflict?

https://github.com/auth0/auth0-vue/blob/main/src/client.proxy.ts#L115-L122

**Comment by frederikprijck** [comment]
@Soviut Yup, I noticed this as well late last night, which is why I mentioned on the issue that I believe we should probably do some more investigation here to ensure we properly address this.

If you don't mind, I would love to dive into this on monday as well, to ensure I am fully aware of what's happening and what we should do. I will circle back to your PR with more info.

File: __tests__/plugin.test.ts [hunk 1]
```diff
@@ -225,7 +225,7 @@ describe('Auth0Plugin', () => {
     expect(handleRedirectCallbackMock).not.toHaveBeenCalled();
 
     return flushPromises().then(() => {
-      expect(replaceStateMock).toHaveBeenCalled();
+      expect(replaceStateMock).not.toHaveBeenCalled();
```
Comment (frederikprijck) [inline __tests__/plugin.test.ts:228]: This test and implementation was actually incorrect and it should not call replaceState when not handling the redirectCallback.

File: playground/router.ts [hunk 2]
```diff
@@ -32,6 +28,6 @@ export function createRouter({ client_id, domain, audience }: any) {
         beforeEnter: authGuard
       }
     ],
-    history: createWebHashHistory()
+    history: createWebHistory()
```
Comment (frederikprijck) [inline playground/router.ts:31]: Moved the playground to use history mode as I believe that should be the default.

File: src/client.proxy.ts [hunk 3]
```diff
@@ -108,20 +107,7 @@ export const createAuth0ClientProxy = (
...
-      const target = appState?.target ?? '/';
-
-      const router = globalProperties.$router as Router;
-
-      if (router && router.push) {
-        router.push(target);
-      }
-
-      return result;
+      return __proxy(() => client.handleRedirectCallback<AppState>(url));
```
Comment (frederikprijck) [inline src/client.proxy.ts:110]: We dont need any router.push here

Comment (frederikprijck) [reply src/client.proxy.ts:110]: This fixed https://github.com/auth0/auth0-vue/issues/93

**Comment by frederikprijck** [comment]
Patched the PR to address things differently.
Thanks for the initial fix here @Soviut !

**Review by stevehobbsdev — APPROVED** [review]

**Pull request merged**